src/components/containers/Fold.js

```diff
@@ -50,10 +58,32 @@ export default class Fold extends Component {
         </div>
       );
 
+    const renderTypeIcon = type => {
+      switch (type) {
+        case 'scatter':
+          return <PlotScatterIcon />;
+        case 'line':
+          return <PlotLineIcon />;
+        case 'bar':
+          return <PlotBarIcon />;
+        case 'pie':
+          return <PlotPieIcon />;
+        case 'area':
+          return <PlotAreaIcon />;
+        default:
+          return <PlotScatterIcon />;
```

Comment:
can we make the default icon something else but scatter?
there's maybe 20-ish trace types in plotly.js, that makes for a long switch statement, could we write this in a more dynamic way..

Reply:
I know.. 

I wonder if something like this is possible -

```
const iconName = `Plot${c(type)}Icon`;
return React.createElement(iconName, null, null);
```

c(string) capitalizes the first letter of the type.

Reply:
It's pretty easy to get all the Plotly traces loaded in a particular instance. See https://github.com/plotly/react-plotly.js-editor/blob/master/src/components/fields/TraceSelector.js#L15 for an example.

Or from a console with a Plotly on window do something like 
```
var schema = Plotly.PlotSchema.get()
console.log(Object.keys(schema.traces))
```

Reply:
(for example https://plot.ly/create)

Reply:
yeah I am thinking about implementing something like:

```
import Icons, {otherIcons} from 'plotly-icons'

...

const renderPlotIcon = (type) => {

    const iconName = `Plot${c(type)}Icon`;

    return Icons[iconName] || defaultIcon;

}

```

Thoughts?

Reply:
I think its a good idea to have a map of icon names that link to the components.
And I think that map should be a constant in the react-plotly.js-editor

That map could be generated through Plotly.PlotSchema.get(), like Ben pointed out, and then we could dynamically check if we have a an icon for that plotly.js plot type:
require('plotly-icons')['PlotScatterIcon'] // we get the icon
require('plotly-icons')['PlotScatterIco'] // undefined

If we don't have an icon, we'd use a default one.

Reply:
What would the map do that 
```
const iconName = `Plot${c(type)}Icon`;
```
isn't?

Reply:
If the plot-icons have a strict format for using the trace names and we have the trace names in editor then 
```
 const iconName = `Plot${c(type)}Icon`;

 return Icons[iconName] || defaultIcon;
```
would work right?
Then if plotly.js adds a new trace type (happens every other month) we add the icon in plot-icons and that's it (aside from bumping the plot-icon version). No need to modify a map in the editor.

Reply:
I'm talking about how we'd generate `Icons`, I'm saying it should be a variable in the editor instead of an export in plotly-icons.
by map I mean that `Icons = {'PlotBarIcon': <PlotBarIcon/>}`
and that it can be dynamic, like you said, but a constant built out in the editor.

Reply:
I don't totally follow. Can you post a pseudo-code snippet that imports the icons from plotly-icons and constructs the `Icon` object?